ReactSkia/core_modules/RSkImageLoader.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+* Copyright (C) 1994-2022 OpenTV, Inc. and Nagravision S.A.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+#include "include/core/SkImage.h"
+
+#include <cxxreact/JsArgumentHelpers.h>
+
+#include "ReactSkia/sdk/CurlNetworking.h"
+#include "ReactSkia/utils/RnsLog.h"
+#include "ReactSkia/components/RSkComponentImage.h"
+#include "RSkImageLoader.h"
+
+using namespace folly;
+
+namespace facebook {
+using namespace xplat;
+using namespace xplat::module;
+
+namespace react {
+
+RSkImageLoaderModule::RSkImageLoaderModule() {
+}
+RSkImageLoaderModule::~RSkImageLoaderModule() {
+  if(imageSizeMap_.size()!=0) {
+    ImageSizemap::iterator itr,it;
+    for(itr = imageSizeMap_.begin(); itr != imageSizeMap_.end(); ++itr) {
+      CurlNetworking::sharedCurlNetworking()->abortRequest(itr->second);
+    }
+    imageSizeMap_.clear();
+  }
+}
+
+auto RSkImageLoaderModule::getConstants() -> std::map<std::string, folly::dynamic> {
+  return {};
+}
+
+auto RSkImageLoaderModule::getMethods() -> std::vector<Method> {
+  return {
+      Method(
+          "getSize",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock){
+            getImageSize(jsArgAsString(args,0),resolveBlock,rejectBlock);
+          }),
+      Method(
+          "getSizeWithHeaders",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            getImageSizeWithHeaders(jsArgAsString(args, 0), jsArgAsString(args, 1), resolveBlock, rejectBlock);
+          }),
+      Method(
+          "prefetchImage",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            prefetchImage(jsArgAsString(args, 0), resolveBlock, rejectBlock);
+          }),
+      Method(
+          "queryCache",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            queryCache(jsArgAsArray(args, 0), resolveBlock, rejectBlock);
+          }),
+  };
+}
+
+std::string RSkImageLoaderModule::getName() {
+  return "ImageLoader";
+}
+
+void RSkImageLoaderModule::getImageSize(std::string uri, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+  auto sharedCurlNetworking = CurlNetworking::sharedCurlNetworking();
+  std::shared_ptr<CurlRequest> remoteCurlRequest = std::make_shared<CurlRequest>(nullptr,uri,0,"GET");
```

Comment:
Should we also check if there is image data for the requested url available in imagedecodrdcache. If not available, then send request for network .

Reply:
Yes. Also, Is it possible to call getImage size for local images ? or  is it always for remote images ?

Reply:
yes. it is always for remote images.